### ISSUE TITLE
Application configuration issues

### ISSUE THREAD

**Issue by adriatic** [monterey-framework/monterey#370]
### Intention

**I am initiating a discussion on "how to use monterey to configure existing applications"**

**Comment by adriatic** [comment]
### Introduction

Monterey is expected to be able to manage Aurelia applications regardless of which module loader is being used by the application (**[SystemJS](https://github.com/systemjs/systemjs)**, **[RequireJS](http://requirejs.org/)**, **[Webpack](https://webpack.github.io/)** or **[WHATWG](https://whatwg.github.io/loader/)**).

___being written___

**Comment by adriatic** [comment]
### Current situation

Current 0.4.4 version of Monterey can scaffold a RequireJS-application, build it and run it. 

![image](https://cloud.githubusercontent.com/assets/2712405/19157262/bd0e1152-8bb2-11e6-99ca-9ccb6fa7765c.png)


![image](https://cloud.githubusercontent.com/assets/2712405/19157340/01ffc8d2-8bb3-11e6-9709-4a9f1a2bf87a.png)

**Comment by adriatic** [comment]
## Proposal for Next step

___At this point things have the tendency to become fuzzy, as everything that follows should be expressed as a huge `if - then- else -if ...`  statement that is too difficult to read more than two levels deep. To avoid this trap, I will sketch a "linear path" that should lead to the solution we should reach first and which will give is more insight for next steps.___

### Extend Monterey to support adding third party modules (including our bridges)

#### Assumptions:
1. People are less concerned about which module loader is being used, as long as they do not to directly handle module loader's configuration and as long as the application works.

2. Offering Monterey users support for using our bridges serves our team the best - and the context should be rich enough to result with very useful feedback.

3. Creating the GUI support for this feature (including the definition of pre and post processing tasks) should be too difficult. We could (and should) use the fork of the [ai-plugin](https://github.com/kristianmandrup/ai-plugin) as a good test for the value of pursuing the needed relationship needed to decide whether to rely on this tool.

4. We could use this context to develop and test the initial prototype of [Online API to get all versions of packages](https://github.com/monterey-framework/monterey/issues/361)

**Comment by adriatic** [comment]
### Definition of the steps needed to support adding third party modules (including our bridges)

**Comment by adriatic** [comment]
Issues to be discussed before continuing the section immediately above:

1. I propose **not to increase the CLI drive scaffolding dialog** defined in **[Aurelia-CLI](https://github.com/aurelia/cli)** - not until Aurelia-CLI adds more features in the scaffolding phase. As a consequence **everything** we are adding to Monterey to [support adding third party modules](https://github.com/monterey-framework/monterey/issues/370#issuecomment-252008295) should be happening once the initially scaffolded app is built and run (as this gives us the access to the Task manager, Workflow manager and other tools that will (likely) have to be added.

2. We may need to re-use parts of the Aurelia-CLI scaffolding code that handles the creation and management of the content of **[`aurelia-project`](https://github.com/aurelia-ui-toolkits/cm-bridges/tree/master/original/aurelia_project)** folder. This should be done by "sub-classing" the original code, so we continue using the most current Aurelia-CLI code in the process of adding third party modules / bridges after the initial scaffolding.

**Comment by JeroenVinke** [comment]
> Clearly we would need to create the GUI that will allow us to execute the tasks like npm install aurelia-kendoui-bridge - leading to the several existing issues:

We can definitely make it easier for people to run `npm install x`. But this would not configure the plugin they installed right?

**Comment by adriatic** [comment]
No, it will not - but as you mentioned, we could add the `pre-install` and `post-install` information into the bridge itself, so we could handle that from Monterey even without having to resort to fork `ai-plugin` and do the work through it. The additional information from our bridge would not be in the way of users who handle the bridge directly - they simply will not ask for that information.
